src/containerapp/azext_containerapp/containerapp_session_code_interpreter_decorator.py

```diff
@@ -0,0 +1,152 @@
+# coding=utf-8
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+
+from knack.log import get_logger
+from typing import Any, Dict
+
+from azure.cli.core.commands import AzCliCommand
+from azure.cli.command_modules.containerapp.base_resource import BaseResource
+from azure.cli.command_modules.containerapp._utils import (_ensure_location_allowed, CONTAINER_APPS_RP)
+from azure.cli.core.commands.client_factory import get_subscription_id
+from ._clients import SessionPoolPreviewClient
+
+from ._models import SessionCodeInterpreterPythonExecution as SessionPoolModel
+from ._client_factory import handle_raw_exception
+
+logger = get_logger(__name__)
+
+class SessionCodeInterpreterPreviewDecorator(BaseResource):
+    def __init__(self, cmd: AzCliCommand, client: Any, raw_parameters: Dict, models: str):
+        super().__init__(cmd, client, raw_parameters, models)
+        self.session_code_interpreter_def = SessionPoolModel
+
```

Comment:
Recommend to init SessionPoolPreviewClient here, so that if this Decorator be moved to Azure-cli for GA, the api-version can be consistent:
`self.session_pool_client = SessionPoolPreviewClient`

Reply:
added :)